libirods_rule_engine_plugin-indexing.cpp

```diff
@@ -411,6 +416,44 @@ namespace {
                 if (auto* p = getValByKey( &obj_inp->condInput, FORCE_FLAG_KW); p != 0) { rm_force_kw = p; }
                 indices_for_rm_coll = get_indices_for_delete_by_query (*_rei->rsComm, obj_inp->objPath, false);
             }
+            else if("pep_api_mod_access_control_post" == _rn ) {
+                irods::indexing::indexer idx{_rei, config->instance_name_};
+                auto it = _args.begin();
+                std::advance(it, 2);
+                if(_args.end() == it) {
+                    THROW(
+                        SYS_INVALID_INPUT_PARAM,
+                        "invalid number of arguments");
+                }
+                try {
+                    const auto* access_ctl = boost::any_cast<modAccessControlInp_t*>(*it);
+                    if (access_ctl) {
+                        const auto &[recursive, level_, userN_, zone_, logical_path] = *access_ctl;
+
+                        namespace fsvr = irods::experimental::filesystem::server;
+                        if (recursive && fsvr::is_collection(*_rei->rsComm,logical_path)) {
+                            idx.schedule_collection_operation(
+                                irods::indexing::operation_type::index,
+                                logical_path,
+                                _rei->rsComm->clientUser.userName
+                                , "::metadata"  //value // -> Empty fields for index_name , index_type ,
```

Comment:
Document that zeroing out the index_name is for enforcing calculation of indexing tag set for every point in the collection hierarchy.